ocaml-lsp-server/src/ocamlformat.ml

```diff
@@ -0,0 +1,239 @@
+open Import
+
+module FileType = struct
+  type t =
+    | Impl
+    | Intf
+    | Name of string
+
+  let to_cmdline_args = function
+    | Impl -> [ "--impl" ]
+    | Intf -> [ "--intf" ]
+    | Name n -> [ Printf.sprintf "--name=%s" n ]
+end
+
+module Input = struct
+  type t =
+    | Stdin of string * FileType.t
+    | File of string
+end
+
+module Output = struct
+  type _ t =
+    | File : string -> unit t
+    | Stdout : string t
+
+  let to_cmdline_args : type o. o t -> string list = function
+    | File s -> [ Printf.sprintf "--output=%s" s ]
+    | Stdout -> []
+end
+
+module Config = struct
+  type t = (string * string) list
+
+  let to_comma_separated_list (conf : t) : string =
+    List.map conf ~f:(fun (k, v) -> Printf.sprintf "%s=%s" k v)
+    |> String.concat ~sep:","
+
+  let of_comma_separated_list (s : string) : t option =
+    String.split_on_char s ~sep:','
+    |> List.fold_right ~init:(Some []) ~f:(fun s state ->
+           match state with
+           | None -> None
+           | Some acc -> (
+             match String.split_on_char s ~sep:'=' with
+             | k :: v :: _ -> Some ((k, v) :: acc)
+             | _ -> None ))
+
+  let of_print_config_output (s : string) : t option =
+    String.split_lines s
+    |> List.fold_right ~init:(Some []) ~f:(fun s state ->
+           match state with
+           | None -> None
+           | Some acc -> (
+             match String.split_on_char s ~sep:' ' with
+             | [ x ]
+             | x :: "(file" :: _ -> (
+               match String.split_on_char x ~sep:'=' with
+               | [ key; value ] -> Some ((key, value) :: acc)
+               | _ -> None )
+             | _ -> None ))
+end
+
+let append_if flag value xs =
+  if flag then
+    value :: xs
+  else
+    xs
+
+let append_opt opt f xs =
+  match opt with
+  | None -> xs
+  | Some x -> f x :: xs
+
+module Options = struct
+  module Profile = struct
+    type t =
+      | Conventional
+      | Default
+      | Compact
+      | Sparse
+      | Ocamlformat
+      | Janestreet
+
+    let to_string = function
+      | Conventional -> "conventional"
+      | Default -> "default"
+      | Compact -> "compact"
+      | Sparse -> "sparse"
+      | Ocamlformat -> "ocamlformat"
+      | Janestreet -> "janestreet"
```

Comment:
I don't think we should include all these options to be honest. In ocamlformat, the plan is to eliminate them anyway.